src/spox/_node.py

```diff
@@ -212,46 +206,40 @@ def pre_init(self, **kwargs):
     def post_init(self, **kwargs):
         """Post-initialization hook. Called at the end of ``__init__`` after other default fields are set."""
 
-    def propagate_values(self) -> dict[str, PropValueType]:
+    def propagate_values(self, input_prop_values: PropDict) -> PropDict:
         """
         Propagate values from inputs, and, if possible, compute values for outputs as well.
         This method is used to implement ONNX partial data propagation - for example so that
         we can reshape into a sum of constant vectors.
         """
         return {}
 
-    def infer_output_types(self) -> dict[str, Type]:
+    def infer_output_types(self, input_prop_values) -> dict[str, Type]:
         """
         Inference routine for output types. Often overriden by inheriting Node types.
 
-        Returns a dictionary of output field names into Types for the respective Vars.
+        Returns a dictionary of output field names into Types for the respective VarInfos.
         """
         return {}
 
-    def inference(self, infer_types: bool = True, propagate_values: bool = True):
+    def inference(self, infer_types: bool = True, input_prop_values={}):
         # Type inference routine - call infer_output_types if required
         # and check if it provides the expected outputs.
-        out_types = self.infer_output_types() if infer_types else {}
+        out_types = (
+            self.infer_output_types(input_prop_values=input_prop_values)
+            if infer_types
+            else {}
+        )
 
-        for key, var in self.outputs.get_vars().items():
+        for key, var in self.outputs.get_var_infos().items():
             if var.type is None:  # If no existing type from init_output_vars
                 # Attempt to use the ones from kwargs, if none then what type inference gave
                 var.type = out_types.get(key)
 
+    def get_output_vars(self, input_prop_values={}):
```

Comment:
I am surprised that no lint was complaining against using a mutable type as a default value. This should read `input_prop_values=None`.